### PR TITLE
Add SHMEM API (v1.3) support

### PR DESCRIPTION
**Pull request by igor-ivanov**
libibprof can be built with Open SHMEM API v1.3 support.
Build:
$./configure CPPFLAGS="-I<shmem installation>/include -DUSE_SHMEM=1"
Use:
env IBPROF_MODE=USE_IBV=0,USE_SHMEM=1
LD_PRELOAD=<libibprof>/libibprof.so oshrun -n 2 hello.out

Limitation:
OpenMPI 1.10+ supports OpenSHMEM API v1.2 at the moment - libibprof can not be used in this case
OpenMPI(master)/OpenMPI2.x+ compatible with OpenSHMEM API v1.3 - libibprof can be used
Nonblocking SHMEM API and alltoall is disabled in libibprof.

Signed-off-by: Igor Ivanov igor.ivanov.va@gmail.com

### CODE REVIEW

**Comment by igor-ivanov** [comment]
@miked-mellanox @yosefe please look at this.

**Comment by Di0gen** [comment]
bot:retest

**Comment by Di0gen** [comment]
bot:retest

**Comment by Di0gen** [comment]
bot:retest

File: README [hunk 1]
```diff
@@ -114,6 +114,7 @@ ibv_poll_cq                    :        350       0.1092       0.0003       0.00
     USE_HCOL - libhcoll
     USE_MXM  - libmxm
     USE_PMIX - libpmix
+    USE_SHMEM - libpmix
```
Comment (igor-ivanov) [inline README]: updated

**Comment by mike-dubman** [comment]
:+1:

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-libibprof-pr/3/ for details (Mellanox internal link).

**Comment by Di0gen** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-libibprof-pr/11/ for details (Mellanox internal link).

**Comment by mike-dubman** [comment]
@igor-ivanov - need to update spec file https://github.com/mellanox-hpc/libibprof/blob/master/libibprof.spec.in#L60 to include USE_SHMEM=1 into cppflags

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-libibprof-pr/18/ for details (Mellanox internal link).

**Pull request merged**